nixos/modules/services/web-apps/goatcounter.nix

```diff
@@ -0,0 +1,57 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib) types;
+  cfg = config.services.goatcounter;
+  stateDir = "goatcounter";
+in
+
+{
+  options = {
+    services.goatcounter = {
+      enable = lib.mkEnableOption "Easy web analytics. No tracking of personal data.";
+
+      package = lib.mkPackageOption pkgs "goatcounter" { };
+
+      address = lib.mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = "Web interface address.";
+      };
+
+      port = lib.mkOption {
+        type = types.port;
+        default = 8081;
+        description = "Web interface port.";
+      };
+
+      proxy = lib.mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether Goatcounter service is running behind a reverse proxy. Refer to [documentation](https://github.com/arp242/goatcounter?tab=readme-ov-file#running) for more details.";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.goatcounter = {
+      description = "Easy web analytics. No tracking of personal data.";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart =
+          "${lib.getExe cfg.package} serve -listen ${cfg.address}:${toString cfg.port}" + lib.strings.optionalString cfg.proxy " -tls proxy";
```

Comment:
```suggestion
        ExecStart = lib.escapeShellArgs ([
          (lib.getExe cfg.package)
          "serve"
          "-listen" 
          "${cfg.address}:${toString cfg.port}"
        ] ++ lib.strings.optionals cfg.proxy [
          "-tls" 
          "proxy"
        ]);
```

Reply:
You could also add a `with types; listOf str` option titled `extraArgs` which is concatenated to this list

Reply:
done